common/src/main/java/org/apache/nemo/common/ir/IRDAG.java

```
@@ -216,13 +216,15 @@ private void deleteRecursively(final IRVertex vertexToDelete, final Set<IRVertex
       modifiedDAG = builder.buildWithoutSourceSinkCheck();
     } else if (vertexToDelete instanceof MessageAggregatorVertex || vertexToDelete instanceof TriggerVertex) {
       modifiedDAG = rebuildExcluding(modifiedDAG, vertexGroupToDelete).buildWithoutSourceSinkCheck();
-      final int deletedMessageId = vertexGroupToDelete.stream()
+      final Optional<Integer> deletedMessageIdOptional = vertexGroupToDelete.stream()
         .filter(vtd -> vtd instanceof MessageAggregatorVertex)
-        .map(vtd -> ((MessageAggregatorVertex) vtd).getPropertyValue(MessageIdVertexProperty.class).get())
-        .findAny().get();
-      modifiedDAG.getEdges().stream()
-        .filter(e -> e.getPropertyValue(MessageIdEdgeProperty.class).isPresent())
-        .forEach(e -> e.getPropertyValue(MessageIdEdgeProperty.class).get().remove(deletedMessageId));
+        .map(vtd -> vtd.getPropertyValue(MessageIdVertexProperty.class).orElseThrow(() ->
+            new RuntimeException("MessageAggregatorVertex " + vtd.getId() + " does not have MessageIdVertexProperty.")))
```

Can you throw a designated exception here, instead of using a general one?

---

(it creates another sonarcloud issue)